File: Top 50 Coding Interview Questions/24. Write code to check a String is palindrome or not/Stringispalindromeornot.py
# Function to check if a string is palindrome
def is_palindrome(s):
    start = 0                   # Initialize start index
    end = len(s) - 1            # Initialize end index

    # Traverse the string from both ends
    while start < end:
        # If characters at start and end are not equal
        if s[start] != s[end]:
            return False   # Not a palindrome
        start += 1         # Move start index to the right
        end -= 1           # Move end index to the left

    return True  # The string is a palindrome

# Input the string
s = input("Enter a string: ")

# Check if the string is a palindrome
if is_palindrome(s):
    print("The string is a palindrome.")
else:
    print("The string is not a palindrome.")

File: Top 50 Coding Interview Questions/24. Write code to check a String is palindrome or not/Stringispalindromeornot.cpp
#include <iostream>
#include <string>
using namespace std;

// Function to check if a string is palindrome
bool isPalindrome(string str) {
    int start = 0;               // Initialize start index
    int end = str.length() - 1;  // Initialize end index

    // Traverse the string from both ends
    while (start < end) {
        // If characters at start and end are not equal
        if (str[start] != str[end]) {
            return false;  // Not a palindrome
        }
        start++;  // Move start index to the right
        end--;    // Move end index to the left
    }
    return true;  // The string is a palindrome
}

int main() {
    string str;

    // Input the string
    cout << "Enter a string: ";
    cin >> str;

    // Check if the string is a palindrome
    if (isPalindrome(str)) {
        cout << "The string is a palindrome." << endl;
    } else {
        cout << "The string is not a palindrome." << endl;
    }

    return 0;
}
